x-pack/plugins/security/server/routes/user_profile/get.ts

```diff
@@ -8,26 +8,40 @@
 import { schema } from '@kbn/config-schema';
 
 import type { RouteDefinitionParams } from '../';
+import type { AuthenticatedUserProfile, UserData } from '../../../common/';
 import { wrapIntoCustomErrorResponse } from '../../errors';
 import { createLicensedRouteHandler } from '../licensed_route_handler';
 
 export function defineGetUserProfileRoute({
   router,
+  getSession,
   getUserProfileService,
 }: RouteDefinitionParams) {
   router.get(
     {
-      path: '/internal/security/user_profile/{uid}',
+      path: '/internal/security/user_profile',
       options: { tags: ['access:accessUserProfile'] },
```

Comment:
Get user profile call is still using internal system user. Only the update call has been updated to use current session. I had to update it since it since the internal system user didn't have permission to update other users profile but I think that's a bug. Administrators should be able to change avatar and user settings in stack management and this service and ES should support that use case.

Reply:
Yeah, that's why I also suggested to use a scoped client for `get` as well. When we need to support "admin" functionality we'll add separate endpoints and separate APIs for that purpose.